src/agent/src/device.rs

```diff
@@ -543,6 +564,32 @@ fn update_spec_devices(spec: &mut Spec, mut updates: HashMap<&str, SpecUpdate>)
         }
     }
 
+    // Correct PCI addresses in the environment
+    if let Some(process) = spec.process.as_mut() {
+        for envvar in process.env.iter_mut() {
+            let (name, val) = envvar
+                .split_once('=')
+                .ok_or_else(|| anyhow!("Malformed OCI env entry {:?}", envvar))?;
+
+            if !name.starts_with("PCIDEVICE_") {
```

Comment:
`PCIDEVICE_` is specific to that one plugin but there are also other env vars for the same purpose (e.g., QAT plugin sets `$QAT<id>`). I guess I could move to `PCIDEVICE_` as well but what's the "spec" for that?

Reply:
Alas, there is no standardized way of conveying this information, so the best we can do is fix up those properties we know about.  That said, I believe there was a proposal to standardize `PCIDEVICE_*` across a number of things.  I've lost track of the document about that, I'll see if I can find it.

With the infrastructure that's been added here, it should be relatively simple to add support for extra formats like QAT, as we discover them, but I don't think that's in scope for this initial series.

Reply:
> That said, I believe there was a proposal to standardize `PCIDEVICE_*` across a number of things. 
> I've lost track of the document about that, I'll see if I can find it.

Thanks! I'd be happy to help with the doc and also update my plugin(s) to follow.

Reply:
Is this a public document? If yes can you please share? Thanks!

Reply:
Ok, managed to dig up the document I was thinking of.  It is public, but it doesn't cover quite what I thought it does.  It does attempt to start standardizing device information more than currently, but it's not  really based on `PCIDEVICE_*` environment variables.  It also appears to be at least somewhat specific to network devices.

https://github.com/k8snetworkplumbingwg/device-info-spec/

Reply:
@dgibson Thanks for that, yeah network-specific, we (RH, NV, IBM, Intel, ...) started some time ago a project that my address this, https://github.com/container-orchestrated-devices/container-device-interface

Reply:
@dgibson - could you add a comment in the code with a link to that doc for future reference (to make less likely someone will come along and decide they'd like to change it to `PCI_DEVICE_` or similar ;)

Reply:
First, I'm not sure which of the documents you mean.  But second, neither one actually talks about `PCIDEVICE_` environment variables.  They're both talking about possible new ways of describing devices, which is a great goal, but not useful to interoperating with the SR-IOV plugin we have right now.